eventauth/eventcontent_test.go

```diff
@@ -0,0 +1,36 @@
+package eventauth
+
+import (
+	"encoding/json"
+	"testing"
+)
+
+func TestLevelJSONValueValid(t *testing.T) {
+	var values []levelJSONValue
+	input := `[0,"1",2.0]`
+	if err := json.Unmarshal([]byte(input), &values); err != nil {
+		t.Fatal("Unexpected error unmarshalling ", input, ": ", err)
+	}
+	for i, got := range values {
+		want := i
+		if !got.exists {
+			t.Fatalf("Wanted entry %d to exist", want)
+		}
+		if int64(want) != got.value {
+			t.Fatalf("Wanted %d got %q", want, got.value)
+		}
+	}
+}
+
+func TestLevelJSONValueInvalid(t *testing.T) {
+	var values []levelJSONValue
+	inputs := []string{
+		`[{}]`, `[[]]`, `["not a number"]`, `["0.0"]`,
```

Comment:
Oh, surprising! Why does 0.0 fail? Shouldn't that be 0? Is this for parity with python's `int()` impl?